src/tests/readytorun/crossgen2/Program.cs

```diff
@@ -2228,7 +2228,15 @@ public static int Main(string[] args)
         RunTest("ExplicitlySizedClassTest", ExplicitlySizedClassTest());
         RunTest("GenericLdtokenTest", GenericLdtokenTest());
         RunTest("ArrayLdtokenTests", ArrayLdtokenTests());
-        RunTest("TestGenericMDArrayBehavior", TestGenericMDArrayBehavior());
+
+        // TODO: BUG BUG: allocation of MD arrays is failing on OSX/ARM64
```

Comment:
I looked at adding a small fix/workaround as a part of the change, but i am not familiar with thatr area of JIT. 
It looks like switching to a different helper might be the right soulution anyways. Not sure how involved that could be.

Reply:
I don't think it's a good idea to try to fix JIT or Crossgen2 as part of this particular PR, just please look up or file a bug to do that.

Reply:
Yes, that’s what I figured. We need professionals here :)

Reply:
It's not about you not being enough of a professional, it's just about keeping GitHub history easier to reason about.

Reply:
Yes, that too. I mean - if that was some trivial one-liner fix like in `MAPmmapAndRecord`, it could be worth doing - just to keep things in a known good state. But as fix appear to be more involved it should be a separate change and by someone more familiar with the area.

Reply:
The bug for this is https://github.com/dotnet/runtime/issues/62747. I will update the comment to have a link to the bug.